ehrapy/api/preprocessing/_data_imputation.py

```diff
@@ -3,51 +3,300 @@
 import numpy as np
 import pandas as pd
 from anndata import AnnData
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.experimental import enable_iterative_imputer  # noqa: F401
+from sklearn.impute import IterativeImputer, KNNImputer, SimpleImputer
+from sklearn.preprocessing import OrdinalEncoder
 
+from ehrapy.api._anndata_util import get_column_indices
 
-def replace_explicit(
-    adata: AnnData,
-    replacement: str | int | dict[str, str | int] | tuple[str, str | int] = None,
-    impute_empty_strings: bool = True,
-    copy: bool = False,
-) -> AnnData | None:
+
+def explicit(adata: AnnData, **kwargs) -> AnnData:
     """Replaces all missing values in all or the specified columns with the passed value
 
-    There are several scenarios to cover:
+    There are two scenarios to cover:
     1. Replace all missing values with the specified value. ( str | int )
-    2. Replace all missing values in a subset of columns with the specified value. ( Dict(str: (str, int)) )
-    3. Replace all missing values in a subset of columns with a specified value per column. ( str ,(str, int) )
+    2. Replace all missing values in a subset of columns with a specified value per column. ( str ,(str, int) )
 
     Args:
         adata: :class:`~anndata.AnnData` object containing X to impute values in
-        replacement: Value to use as replacement and optionally keys to indicate which columns to replace.
-        See scenarios above
-        impute_empty_strings: Whether to also impute empty strings
         copy: Whether to return a copy with the imputed data.
+        **kwargs: replacement: Value to use as replacement and optionally keys to indicate which columns to replace.
```

Comment:
This really should not be a kwargs in my opinion. This is a key argument. This function is useless without an explicit `replacement`.